metasynth/dataset.py

```diff
@@ -117,6 +117,9 @@ def from_dataframe(cls,
         else:
             spec = deepcopy(spec)
 
+        if set(list(spec)) - set(df.columns):
+            raise ValueError("Specifications found for column that were not found in the "
+                             f"dataset itself: {set(list(spec)) - set(df.columns)}")
```

Comment:
@Samuwhale do you have a suggestion for a better error message here? I am wondering if this is immediately understandable, but I'm finding it hard to decide on a better error message.

I checked what tidyverse does when loading a csv with a spec with a column that could not be found; it displays a warning (not an error) and it's the following message:
``` 
Warning message:                                                                     
The following named parsers don't match the column names: Pid 
```
I think that is kind of worse so maybe our message is ok 😃

Reply:
Feel free to merge and create a new issue out of this comment